aiida/backends/sqlalchemy/migrations/versions/041a79fc615f_dblog_cleaning.py

```diff
@@ -0,0 +1,182 @@
+# -*- coding: utf-8 -*-
+###########################################################################
+# Copyright (c), The AiiDA team. All rights reserved.                     #
+# This file is part of the AiiDA code.                                    #
+#                                                                         #
+# The code is hosted on GitHub at https://github.com/aiidateam/aiida_core #
+# For further information on the license, see the LICENSE.txt file        #
+# For further information please visit http://www.aiida.net               #
+###########################################################################
+# pylint: disable=invalid-name,no-member,import-error,no-name-in-module
+"""This migration cleans the log records from non-Node entity records.
+It removes from the DbLog table the legacy workflow records and records
+that correspond to an unknown entity and places them to corresponding files.
+
+This migration corresponds to the 0024_dblog_update Django migration.
+
+Revision ID: 041a79fc615f
+Revises: 7ca08c391c49
+Create Date: 2018-12-28 15:53:14.596810
+"""
+# pylint: disable=wrong-import-order
+from __future__ import absolute_import
+from __future__ import print_function
+
+import click
+import sqlalchemy as sa
+from alembic import op
+from six.moves import zip
+
+from aiida import settings
+from aiida.backends.sqlalchemy.utils import dumps_json
+
+# revision identifiers, used by Alembic.
+revision = '041a79fc615f'
+down_revision = '7ca08c391c49'
+branch_labels = None
+depends_on = None
+
+# The values that will be exported for the log records that will be deleted
+values_to_export = ['id', 'time', 'loggername', 'levelname', 'objpk', 'objname', 'message', 'metadata']
+
+
+def get_legacy_workflow_log_no(connection):
+    """ Get the number of the log records that correspond to legacy workflows """
+    return connection.execute('SELECT COUNT(*) FROM db_dblog WHERE (db_dblog.objname '
+                              'LIKE \'aiida.workflows.user.\' || \'%%\')').fetchall()[0][0]
+
+
+def get_unknown_entity_log_no(connection):
+    """ Get the number of the log records that correspond to unknown entities """
+    return connection.execute('SELECT COUNT(*) FROM db_dblog WHERE (db_dblog.objname '
+                              'NOT LIKE \'node.\' || \'%%\') AND (db_dblog.objname '
+                              'NOT LIKE \'aiida.workflows.user.\' || \'%%\')').fetchall()[0][0]
+
+
+def get_serialized_legacy_workflow_logs(connection):
+    """ Get the serialized log records that correspond to legacy workflows """
+    query = connection.execute(
+        'SELECT db_dblog.id, db_dblog.time, db_dblog.loggername, db_dblog.levelname, db_dblog.objpk, '
+        'db_dblog.objname, db_dblog.message, db_dblog.metadata FROM db_dblog WHERE (db_dblog.objname '
+        'LIKE \'aiida.workflows.user.\' || \'%%\')')
+    res = list()
+    for row in query:
+        res.append(dict(list(zip(row.keys(), row))))
```

Comment:
Do you need the intermediate `list`? Also, according to my comment above, you might just zip with `values_to_export` instead